CMakeLists.txt

```diff
@@ -0,0 +1,342 @@
+cmake_minimum_required(VERSION 3.12)
```

Comment:
Would it be possible to make this at least 3.24? (3.24 was released in Aug 2022.)

Reply:
Do you guarantee that everybody uses the newest version?
Please, can you show me a feature of the latest version of cmake for this solution to justify using it?

The chosen version is the minimum that has basic support on `fetchContent` and other cmake `target_*` functions.

Reply:
By way of background:

 * 3.12.0 was released in 2018
 * 3.12.4 (the most recent patch) was released in 2018
 * 3.24.0 was released in 2022
 * 3.24.4 (the most recent patch) was released in 2023
 * 3.27.5 is the most current version

Declaring 3.12 as the minimum version today makes it much more difficult to take advantage of the features CMake has added over the last five years.

3.24 is a compromise of sorts; it's new enough to take advantage of many improvements, but not so new that it requires users to be at the bleeding edge.

In particular: 3.24 adds very useful integration between `FetchContent` and `find_package`. There are many ways for facil, specifically, to take advantage of that integration. But that's just one example; there are many more between 3.12 and 3.24.

On top of that…

What I've seen happen in almost every project I've looked at is that the version declared the first time CMake is added to a project *never changes*. If that become the case here too, it seems odd to, today, stake yourself forever to an already five year old version.

To stretch a metaphor a little too far: it would be a bit like saying, today, "you can't use any C features beyond C99."

Reply:
`cmake_minimum_required(VERSION 3.12...3.24)` - <min> ... <max>